src/Check/Security/D7/ViewsAccess.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * @file
+ * Contains Drutiny\Check\Security\D7\ViewsAccess
+ */
+
+namespace Drutiny\Check\Security\D7;
+
+
+use Drutiny\Check\Security\ViewsAccessBase;
+use Drutiny\Annotation\CheckInfo;
+use Drutiny\Helpers\Serializer;
+
+/**
+ * @CheckInfo(
+ *   title = "Views Access",
+ *   description = "Ensure views meet minimum access permissions.",
+ *   remediation = "Update views to match expected permissions.",
+ *   success = "<code>:num</code> views configured correctly.",
+ *   failure = "<code>:no_access_num</code> view:no_access_plural have no access control set<br/><ul><li>:no_access</li></ul><br/><br/><code>:weak_num</code> view:weak_num_plural have weak permissions<br/><ul><li>:weak</li></ul>",
+ *   exception = "Could not find any views",
+ *   not_available = "No views available",
+ * )
+ */
+class ViewsAccess extends ViewsAccessBase {
+
+  /**
+   * Access all views for the site.
+   *
+   * @return array
+   *   All available views.
+   */
+  protected function getViews() {
+
+    // We assume that permissions will not be set on other types of views and
+    // their permissions will be controlled via the embed method (blocks, panes
+    // etc.).
+    $sql = "SELECT vv.vid, vv.name, vv.description, vv.human_name, vd.id, vd.display_title, vd.display_options, vd.display_plugin
```

Comment:
If it helps, I have written a `runScript` method in the drush controller, to which you can execute raw PHP in the context of the site. This maybe of use here.